src/event/event-type.js

```diff
@@ -96,79 +96,95 @@ const Html5EventType: EventTypes = {
 
 const CustomEventType: EventTypes = {
   /**
-   * Fires when the player enters fullscreen
+   * Fires when the player starts reset operation.
+   */
+  RESET_STARTED: 'resetstarted',
```

Comment:
@dan-ziv why would we need the `STARTED`|`ENDED` semantics?
In both operations you call the `reset`|`destroy' on all player managed objects, e.g. event manager, plugins etc. so everyone has ability to respond in time to the operation.
IMO I think it's enough to have `PLAYER_RESET` and `PLAYER_DESTROYED` events.
WDYT?

Reply:
I wanted to allow full lifecycle support for those operations (same as we did with change media).
when reset/destroy started the player still has its data (state, tracks, poster, source...), so if a certain app/component would like to save something or do some action (UI, routing, god knows what) before reseting/destroying this is the right time. When reset/destroy ended the player is all cleaned up and the component/app can respond accordingly. 
As far as the player is concerned, each will manage it the way he sees fit, but the player has to announce when he starts and when he finishes.

As for the plugins, I think we should do the same there and provide lifecycle methods for plugins:
`beforeReset`
`afterReset`
`beforeDestroy`
`afterDestroy`